core/sql/executor/ExHdfsScan.cpp

```diff
@@ -1378,9 +1423,27 @@ ExWorkProcRetcode ExHdfsScanTcb::work()
   
   return WORK_OK;
 }
+void ExHdfsScanTcb::setR2ColumnNull(Int32 colidx)
+{
+  Lng32 neededColIndex = 0;
+  Attributes * attr = NULL;
+  ExpTupleDesc * asciiSourceTD = hdfsScanTdb().workCriDesc_->getTupleDescriptor(hdfsScanTdb().asciiTuppIndex_);
+  for (Lng32 i = 0; i <  hdfsScanTdb().convertSkipListSize_; i++)
+  {
+    if (hdfsScanTdb().convertSkipList_[i] > 0)
+    {
+      attr = asciiSourceTD->getAttr(neededColIndex);
+      neededColIndex++;
+	  if(attr->getOffset() == colidx)
+	{
+		*(short *)&hdfsAsciiSourceData_[attr->getNullIndOffset()] = -1;
+	}
+    }
+  }
+}
```

Comment:
I wonder if we need to change it to NULL at line no 1585 when convertExpr() returns an error  if the column is nullable and the cqd is set to convert to null. It is not clear to me the need for other changes.